### PR TITLE
task 6 add comments to article page

### PR DESCRIPTION
**Pull request by amyf211**

### CODE REVIEW

File: news-buddy/src/components/ArticlesList.jsx [hunk 1]
```diff
@@ -2,17 +2,19 @@ import { useState, useEffect } from "react"
...
 import Loading from "./Loading"
 
-function ArticlesList({isLoading, setIsLoading}) {
+function ArticlesList() {
     const [articlesList, setArticlesList] = useState([])
+    const [isLoading, setIsLoading] = useState(true)
 
     useEffect(() => {
+        setIsLoading(true)
         fetch('https://newsbuddy-f5h1.onrender.com/api/articles')
```
Comment (nc-saima-khanom) [inline news-buddy/src/components/ArticlesList.jsx:11]: We could extract out this logic into a separate api.js file which can house all of our api calls :)

File: news-buddy/src/components/CommentsList.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from "react"
+import CommentCard from "./CommentCard"
+import Loading from "./Loading"
+
+function CommentsList({ article_id }) {
+    const [commentsList, setCommentsList] = useState([])
```
Comment (nc-saima-khanom) [inline news-buddy/src/components/CommentsList.jsx:6]: lets also add a loading state here for the comments load

File: news-buddy/src/components/CommentsList.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    useEffect(() => {
+        fetch(`https://newsbuddy-f5h1.onrender.com/api/articles/${article_id}/comments`)
```
Comment (nc-saima-khanom) [inline news-buddy/src/components/CommentsList.jsx:9]: we can also add this the separate api.js file housing all the api requests

File: news-buddy/src/components/CommentsList.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        .then((data) => {
+            setCommentsList(data)
+        })
+    })
+
+return(
+    <section id="comments-list">
+        <h3>Comments:</h3>
+        <ul>
+            {commentsList.map((comment) => {
```
Comment (nc-saima-khanom) [inline news-buddy/src/components/CommentsList.jsx:20]: nice!

File: news-buddy/src/components/ArticlesList.jsx (hunk 1 above)
Comment (amyf211) [reply news-buddy/src/components/ArticlesList.jsx:11]: Hi Saima! Do you mean the fetch inside the useEffect could be extracted out?

**Pull request merged**